### ISSUE TITLE
ImportError in APEX package while following the Installation guide

### ISSUE THREAD

**Issue by pcanas** [facebookresearch/vissl#279]
## 📚 VISSL Documentation

I have tried all the different installation options mentioned in the [Intallation guide](https://vissl.readthedocs.io/en/v0.1.5/installation.html) but they all converge to the same issue when I try to import the `apex` package:

```
File "/home/[...]/.local/lib/python3.7/site-packages/apex/__init__.py", line 13, in <module>
    from pyramid.session import UnencryptedCookieSessionFactoryConfig
ImportError: cannot import name 'UnencryptedCookieSessionFactoryConfig' from 'pyramid.session' (unknown location)
```

I tried looking for similar issues but I did not find a clear answer. Also, I was wondering if the library is used at all since I cannot see its use during the tutorials.

**Comment by prigoyal** [comment]
thank you @pcanas , for us to help further, can you share how you tried to install ? 
- using pip via source
- using conda via source
- using pre-built pip packages
- using pre-built conda packages

**Comment by prigoyal** [comment]
@bottler , I have had one more similar report of this issue from a different user and in their case, the installation used pip from source. Can you help look into it?

**Comment by bottler** [comment]
I am happy to have a chance to explain something about the apex name clash.

There is a package called `apex` on PyPI, which you can get with `pip install pypi`. It describes itself as `Pyramid toolkit to add Velruse, Flash Messages,CSRF, ReCaptcha and Sessions`, and its most recent version is `0.9.10dev`, from 2013. It depends on something called `pyramid`. It is nothing to do with NVIDIA apex (which is used by VISSL) and is not a dependency of VISSL. I think if you want to use VISSL then actually having it installed is not a good idea, because it may break things. VISSL expects imports from `apex` to be NVIDIA apex. 

NVIDIA apex cannot be installed with `pip install apex`. One of the ways we suggest to install it in INSTALL.md is with a command like `pip install -f https://dl.fbaipublicfiles.com/vissl/packaging/apexwheels/<some tag>/download.html apex`. It is possible that if the value of `<some tag>` is invalid or not one we support or is for a version of python which isn't what you have then this command with default back to getting apex from PyPI. This could be prevented by recommending apex's dependencies be installed first and then the main command could use pip's `--no-index` option. So the recommendation might need to be
```
pip install cxxfilt>=0.2.0 tqdm>=4.28.1 numpy>=1.15.3 PyYAML>=5.1 pytest>=3.5.1
pip install --no-index -f https://dl.fbaipublicfiles.com/vissl/packaging/apexwheels/<some tag>/download.html apex
```
Another, more dirty option might be to fix the version 
```
pip install -f https://dl.fbaipublicfiles.com/vissl/packaging/apexwheels/<some tag>/download.html apex=0.0
```
I suggest that anyone with errors referring to pyramid is likely to have the wrong apex installed.

**Comment by prigoyal** [comment]
thank you @bottler .

@pcanas , I have just updated the tutorials to follow `pip install -f https://dl.fbaipublicfiles.com/vissl/packaging/apexwheels/{version_str}/download.html apex` https://colab.research.google.com/drive/1Rt3Plt3ph84i1A-eolLFafybwjrBFxYe?usp=sharing and our [Install guide](https://github.com/facebookresearch/vissl/blob/master/INSTALL.md#step-2-install-pytorch-opencv-and-apex-pip) is up-to-date to reflect the proper way to install apex. 

Please let us know if this doesn't solve the issue! :)

**Comment by pcanas** [comment]
Hi @prigoyal thanks for your support! After learning about `apex` clash, I just created my own Python environment and only installed NVIDIA's package. Everything works now :)

Thank you for sharing this insight @bottler

**Comment by prigoyal** [comment]
that's great to hear! closing this issue then :)

**Comment by kulkarnivishal** [comment]
@prigoyal the fix doesn't work for pytorch 1.10.1+cu111 and version_str is sliced incorrectly -- py38_cu111_pyt11. 
would like to know which apex versions are supported cuz py38_cu111_pyt1101 doesn't work

**Comment by rtaori** [comment]
Please provide an update on this. I'm using pytorch 1.12.0, cuda 11.3, python 3.8 and cannot find the correct installation string
